Codecademy/gamut
# feat(Anchor): Add icon to Anchor

### Overview
1. Adding an icon to the Anchor component. 
2. Applying logic from InlineIconButton's `getButtonContent()` function and refactoring that to work across Buttons, Badges, and Anchors. 
3. Added in some tests for Anchor (rendering purposes, can remove if it's unnecessary)
4. Updated figma link for icons (regular and mini)





### PR Checklist

- [x] Related to designs: https://www.figma.com/design/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=60-3&m=dev
- [x] Related to JIRA ticket: [GM-704](https://codecademy.atlassian.net/browse/GM-704)

#### Testing Instructions



1. Go to `/?path=/docs/typography-anchor--anchor-with-icon#adding-an-icon` and check that the icon loads next to the Anchor (both left and right examples) 
2.  Go to `/?path=/docs/atoms-badge-badge--sample-icon#leading-icon` and check that the leading icon loads and looks centered
5. Go to `/?path=/docs/atoms-button--inline-icon#inline-icon` and check that the icons load properly
---  
6. Go to the Mono PR's Preview
7. Log in and navigate to the Profile (`profiles/<userID>`) page
8. Check the 'Latest Courses', if you don't have any courses, you should see: 
9. <img width="340" alt="image" src="https://github.com/Codecademy/gamut/assets/25854220/a3c33f0e-c6f0-49d7-bfbc-cd5d7fd7965a">
10.  If there are enrolled courses, check against the Streaks section, check that the streaks link renders an Icon 
11. <img width="411" alt="image" src="https://github.com/Codecademy/gamut/assets/25854220/e7ea7bdd-7cdb-4d06-a78e-62d64375ec96">
 ---
12. Go to the Monolith PR's Preview
13. Go to the Signup page (`/register`) and check the bottom text's Terms & Conditions link and check that the icons render
14. <img width="369" alt="image" src="https://github.com/Codecademy/gamut/assets/25854220/5273af8d-9e40-4cfb-8955-e4c38576a4a9">
15. NOTE: these visual tests will be removed before merging
16. ...
17. Profit!?



#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/37833)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Mono PR       | [Mono PR](https://github.com/codecademy-engineering/mono/pull/6483)  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |




[GM-704]: https://codecademy.atlassian.net/browse/GM-704?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
